src/Sentry.AzureFunctions.Worker/.editorconfig

```diff
@@ -0,0 +1,5 @@
+# https: //docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference
+[*.cs]
+
+# Justification: Azure Functions (single thread) project
+dotnet_diagnostic.CA2007.severity = none
```

Comment:
I'm not so sure about this.

`ConfigureAwait(false)` (CA2007) isn't about being single threaded or not (and I'm not sure that's accurate anyway).

Rather, `ConfigureAwait` is about _synchronization context_.  The `false` is assigned to the `continueOnCapturedContext` parameter,  which defaults to `true`.  However, if there's no synchronization context, then it doesn't matter what this setting is.

For example, ASP.NET Core applications don't have a synchronization context, but ASP.NET Framework and desktop applications do.  The usual advice for _libraries_ is to always set `ConfigureAwait(false)`.

As for Azure Functions, I can find very little concrete information about this with regard to the isolated worker model.  I only found a bit about [durable functions](https://learn.microsoft.com/azure/azure-functions/durable/durable-functions-code-constraints#net-threading-apis) having a synchronization context, and requiring `ConfigureAwait(true)` (or omitting for the default).  Even then, this would apply to the orchestrator functions themselves, not necessarily any libraries they are using.

One thing that took me a long time to wrap my head around is that the placement of `ConfigureAwait` affects the function in a way that is not what you might think it is.

```csharp
var result = await DoSomethingAsync().ConfigureAwait(false);
```

You might think it's interpreted like this:

```csharp
var result = (await DoSomethingAsync()).ConfigureAwait(false);
```

But actually it's like this:

```csharp
var result = await (DoSomethingAsync().ConfigureAwait(false));
```

So the `ConfigureAwait` is affecting the `await` itself - which controls the step in the state machine at this particular await.  Thus in the following:

```csharp
var result = await DoSomethingAsync();

async Task<T> DoSomethingAsync()
{
    return await DoSomethingElseAsync().ConfigureAwait(false);
}
```

... there is an implicit `ConfigureAwait(true)` in the first call, so the `ConfigureAwait(false)` in the inner function only applies to the `await` in _that_ function.  This is why we set false in library code - because we don't want to create any _additional_ synchronization requirements.  It doesn't affect the synchronization of the caller.

I can't think of why someone would _need_ synchronization context in an Azure functions isolated worker, but I can see that there might still be one present.  Perhaps only with a .NET Framework implementation?  Not sure.  Still - as a library, we should still set `ConfigureAwait(false)` as to not create additional synchronization requirements.

Reply:
Thank you for the elaborate comment. I was unaware of how `await` is applied with `ConfigureAwait()` in place. 
For Functions, I remember from the in-process SDK that there was no synchronization context. Isolated Worker SDK continued that. [This answer](https://stackoverflow.com/questions/55542016/is-configureawaitfalse-needed-beneficial-when-awaiting-async-calls-in-azure-fu) from Stephen Cleary mentions what I remembered. I can also try and get a concrete answer for the Isolated Worker SDK to confirm it.

Reply:
Raised a [question](https://github.com/Azure/azure-functions-dotnet-worker/issues/1520).

Reply:
Yeah, I'd still like to err on the side of safety and consistency.  For us that means, anything under `/src` should be using `ConfigureAwait(false)`.  It sounds like it won't hurt to have it either way.

Thanks for raising the question to confirm though. 👍